kombu/asynchronous/http/__init__.py

```diff
@@ -16,8 +16,8 @@ def Client(hub=None, **kwargs):
 def get_client(hub=None, **kwargs):
     """Get or create HTTP client bound to the current event loop."""
     hub = hub or get_event_loop()
-    try:
-        return hub._current_http_client
-    except AttributeError:
+    assert hub is not None
```

Comment:
I am not sure this is equivalent:
- if `hub` is `None` an `AssertionError` will now be raised, instead of `AttributeError` here.
- if assertions are disabled (`python -O` flag), this code will raise an `AttributeError` on line 22 now.

As far as I can see this adds failure cases, instead of simplifying operations.

Reply:
@georgepsarakis I can rewrite it as:
```py
if hub is None:
   raise RuntimeError('hub is not initialized')
```
(I used `assert` since I noticed there's many `assert`s in the code whenever an assumption was made.)

The larger issue is that this code absolutely expects hub to be initialized. The `except AttributeError` was intended to handle `_current_http_client` being unset, but it ended up unintentionally handling `None._current_http_client`. Of course the handling code was not designed to create a hub (I'd assume it's not the task of `get_client` but I'm not very familiar with the code).